### PR TITLE
Return pip binary without -m flag when using onedir

### PR DESCRIPTION
**Pull request by xeacott**
### What does this PR do?
When using the onedir inside dist/run from pop-build, pip is provided and therefore cannot handle the -m flag. The pip module needs to handle this case.

### What issues does this PR fix or reference?
Fixes:
https://github.com/saltstack/release-planning/issues/239

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog
- [x] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by s0undt3ch — COMMENTED** [review]
Should we keep functionality as it was and provide an alternate logic if the app is frozen and/or has the `sys._MEIPASS` attribute?

**Comment by xeacott** [comment]
@s0undt3ch I'm open to taking this in a different direction. The heart of the issue is that the executable that sys.executable is reporting is *not* python, just something else; in the case of the onedir it's `./run`. So the `-m` that gets passed around isn't going to a python binary but rather the pip binary.

**Comment by s0undt3ch** [comment]
Sure, and we should handle that. Draw your own conclusions and solution _when_ running under pyinstaller.
If it's not running under pyinstaller, then the logic should remain as it is today.

Makes sense?

**Comment by s0undt3ch** [comment]
```
if under_pyinstaller:
     ...
else:
    ...
```

File: salt/modules/pip.py [hunk 1]
```diff
@@ -144,14 +144,30 @@ def _clear_context(bin_env=None):
...
 
+def _check_bundled():
+    """
+    Gather run-time information to indicate if we are running from source or bundled.
+    """
+    frozen = False
+    if getattr(sys, "frozen", False) and hasattr(sys, "_MEIPASS"):
+        frozen = True
+    return frozen
+
```
Comment (s0undt3ch) [inline salt/modules/pip.py:155]:
```
ef _check_bundled():
    """
    Gather run-time information to indicate if we are running from source or bundled.
    """
    if getattr(sys, "frozen", False) and hasattr(sys, "_MEIPASS"):
        return True
    return False
```
One less local variable

File: tests/unit/modules/test_pip.py [hunk 2]
```diff
@@ -38,6 +38,11 @@ def test__pip_bin_env_no_bin_env(self):
         ret = pip._pip_bin_env(None, None)
         self.assertIsNone(ret)
 
+    def test__pip_bin_bundled_app(self):
+        sys_info = hasattr(sys, "sys._MEIPASS")
```
Comment (s0undt3ch) [inline tests/unit/modules/test_pip.py:42]: `sys_info = hasattr(sys, "_MEIPASS")`?

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**